system-contracts/contracts/libraries/SystemContractsCaller.sol

```diff
@@ -114,6 +113,28 @@ library SystemContractsCaller {
                 success := call(msgValueSimulator, callAddr, value, to, farCallAbi, forwardMask, 0)
             }
         }
+        if (success && returnDataCopy) {
```

Comment:
It is a bit of a nit, but copying into slice of `0..returndatalen` is a very rare use-case (and unless the memory is manually managed like in EvmSimulator) is very dangerous.

I'd accept the same three params as `returndatacopy` does and use those inside `returndatacopy`

Reply:
fixed